cmd/kubelet/app/options/options.go

```diff
@@ -126,15 +126,6 @@ type KubeletFlags struct {
 	ExperimentalNodeAllocatableIgnoreEvictionThreshold bool
 	// Node Labels are the node labels to add when registering the node in the cluster
 	NodeLabels map[string]string
-	// lockFilePath is the path that kubelet will use to as a lock file.
-	// It uses this file as a lock to synchronize with other kubelet processes
-	// that may be running.
-	LockFilePath string
-	// ExitOnLockContention is a flag that signifies to the kubelet that it is running
-	// in "bootstrap" mode. This requires that 'LockFilePath' has been set.
-	// This will cause the kubelet to listen to inotify events on the lock file,
-	// releasing it and exiting when another process tries to open that file.
-	ExitOnLockContention bool
```

Comment:
Removing them from the CLI flags at the same time that is added to the config file makes it a little harder to upgrade/downgrade.

IIUC the [deprecation policy for flags][link], rule 5b and 6 applies. Therefore we should:
 * Just warn on these flags being used (if we aren't doing it already)
 * Add the config option so users can migrate now
 * 6 months or 2 releases later remove the CLI flags

Am I missing something? Was this process already started or something?

[link]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli

Reply:
CLI flags are still bound for these options, see further down in the same file

Reply:
Oops. Sorry for the noise, then! :)